services/api/src/resources/deploytargetconfig/resolvers.ts

```diff
@@ -79,6 +84,73 @@ export const getDeployTargetConfigsByDeployTarget: ResolverFn = async (
   return withK8s;
 };
 
+export const updateEnvironmentDeployTarget: ResolverFn = async (
+  root,
+  input,
+  utils
+) => {
+
+  const { environment, deployTarget } = input;
+  const { sqlClientPool, hasPermission, userActivityLogger } = utils;
+  let environmentObj =  await environmentHelpers(
+    sqlClientPool
+  ).getEnvironmentById(parseInt(environment));
+
+
+  await hasPermission('project', 'update', {
+    project: environmentObj.project
+  });
+
+  const deployTargets = await getDeployTargetConfigsByProjectId(null, {project: environmentObj.project}, utils);
+
+  let matchesRule = false;
+  let ruleMatch = null;
+  if(deployTargets.length > 0) {
+    let endLoop = false;
+    for(let i = 0; i < deployTargets.length && !endLoop; i++) {
+      let branchTarget = deployTargets[i].branches;
+      switch(branchTarget) {
+        case(undefined):
+        case(null):
+        case(true):
+        case(false):
+          // if any of these before a match, we're actually done
+          // because we don't have a _specific_ rule
+          endLoop = true;
+          continue;
+        break;
+        default:
+          let branchRegex = new RegExp(branchTarget);
+          if(branchRegex.test(branchTarget)) {
+            matchesRule = deployTargets[i].deployTarget == deployTarget;
+            endLoop = true;
+          }
+          //if there's no match for this first target,
+          //we continue on since there may be one at lighter weights
+          continue;
+      }
+    }
+  }
+
+  if(matchesRule == false) {
+    throw new Error("Cannot change deploy target without matching Deploy Target rule");
+  }
+
+  await query(
+    sqlClientPool,
+    EnvironmentSql.updateEnvironment({
+      id: environment,
+      patch: {
+        openshift: deployTarget,
+      }
+    })
+  );
+
```

Comment:
can we get an activity logger in here to report that the environment had it's deployTarget updated?

Reply:
there we go.